src/03-filter.js

```diff
@@ -6,28 +6,46 @@
   Keep in mind that your functions must still have and use a parameter for accepting all songs.
 */
 const exampleSongData = require("../data/songs");
+const { printAllSongTitles } = require("./01-forEach");
 // Do not change the line above.
 
 /**
  * Returns an array of all songs by the artist "Saib".
  * @param {Object[]} songs - An array of songs. See the song data for more.
  * @returns {Object[]} An array of objects.
  */
```

Comment:
Filter - the callback for 'filter' should return a boolean reflecting the expression you want to filter by. The condition for your if statements inside the filter callback are what you should have returned from the callback, making your code a lot more concise. The reason it worked is that if the return value from the callback is not 'true' or 'false' JS will evaluate it's truth or falsy nature, and a "song" would have a truth value while undefined (which is return when nothing else is returned from a function) would have a falsy value. Let me know if you need further explanation or clarifications about this